src/clients/UBAClient/UBAClientWithRefresh.ts

```diff
@@ -47,14 +46,64 @@ export class UBAClientWithRefresh extends BaseUBAClient {
         this.spokePoolClients,
         this.chainIdIndices,
         this.tokens,
-        this.hubPoolClient.latestBlockNumber ?? 0,
         forceClientRefresh,
-        this.relayerConfiguration,
         this.maxBundleStates
       )
     );
   }
 
+  /**
+   * Compute the realized LP fee for a given amount.
+   * @param hubPoolTokenAddress The L1 token address to get the LP fee
+   * @param depositChainId The chainId of the deposit
+   * @param refundChainId The chainId of the refund
+   * @param amount The amount that is being deposited
+   * @param hubPoolClient A hubpool client instance to query the hubpool
+   * @param spokePoolClients A mapping of spoke chainIds to spoke pool clients
+   * @returns The realized LP fee for the given token on the given chainId at the given block number
+   */
+  public async computeLpFee(
+    hubPoolBlockNumber: number,
+    amount: BigNumber,
+    depositChainId: number,
+    refundChainId: number,
```

Comment:
Do we want to name it `refundChainId` ? I think it's dangerous for third-party integrators who might end up computing invalid LP fees when filling relays.